shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/federate/schema/table/generator/FederateExecutionSQLGenerator.java

```diff
@@ -35,14 +37,17 @@
     
     private final int[] projects;
     
+    private final List<String> columnNames;
+    
     /**
      * Generate sql.
      *
      * @param table table
      * @return sql
      */
     public String generate(final String table) {
-        // TODO
-        return String.format("SELECT * FROM %s", table);
+        // TODO generate sql with filters
+        String project = null == projects ? "*" : Arrays.stream(projects).mapToObj(columnNames::get).collect(Collectors.joining(", "));
```

Comment:
Considering `projects` is for SQL optimization, at this moment, we can consider returning the result only with `Arrays.stream(projects).mapToObj(columnNames::get).collect(Collectors.joining(", ")`. How do you think?

Reply:
> Considering `projects` is for SQL optimization, at this moment, we can consider returning the result only with `Arrays.stream(projects).mapToObj(columnNames::get).collect(Collectors.joining(", ")`. How do you think?

@tristaZero Good suggestion, I will replace all columns with `*`.